packages/flutter/test/cupertino/date_picker_test.dart

```diff
@@ -11,6 +11,120 @@ import 'package:flutter_test/flutter_test.dart';
 const Offset _kRowOffset = Offset(0.0, -32.0);
 
 void main() {
+  testWidgets('time picker scrolls to 00:00:00 after reset', (WidgetTester tester) async {
+    const Duration animationTime = Duration(milliseconds: 200);
+    final CupertinoTimerPickerController controller = CupertinoTimerPickerController(
+        resetAnimationDuration: animationTime,
+    );
+    Duration duration;
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: Padding(
+          padding: const EdgeInsets.all(40.0),
+          child: Column(
+            children: <Widget>[
+              Container(
+                height: 300,
+                child: CupertinoTimerPicker(
+                  controller: controller,
+                  onTimerDurationChanged: (Duration newDuration) { duration = newDuration; },
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+
+    await tester.drag(find.text('0').first, _kRowOffset * 2);
```

Comment:
Can you add a comment explaining what these low level actions are attempting to achieve?